File: isma-compiler/hsm-core/src/main/java/ru/nstu/isma/core/hsm/hybrid/HMState.java
package ru.nstu.isma.core.hsm.hybrid;

import ru.nstu.isma.core.hsm.var.HMVariableTable;

import java.io.Serializable;

/**
 * Created by Bessonov Alex
 * Date: 24.10.13
 * Time: 23:32
 */
public class HMState implements Serializable {
    protected String code;

    protected HMVariableTable variables;

    public HMState(String code) {
        variables = new HMVariableTable();
        this.code = code;
    }

    public HMVariableTable getVariables() {
        return variables;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        HMState hmState = (HMState) o;

        return code.equals(hmState.code);
    }

    @Override
    public int hashCode() {
        return code.hashCode();
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/hybrid/HMState.kt
package ru.nstu.isma.hsm.hybrid

import ru.nstu.isma.hsm.`var`.HMVariableTable
import java.io.Serializable

/**
 * Created by Bessonov Alex
 * Date: 24.10.13
 * Time: 23:32
 */
open class HMState(code: String) : Serializable {
    var code: String
    var variables: HMVariableTable
        protected set

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val hmState = other as HMState
        return code == hmState.code
    }

    override fun hashCode(): Int {
        return code.hashCode()
    }

    init {
        variables = HMVariableTable()
        this.code = code
    }
}